### PR TITLE
fix: only set model name if the model really changes and refactor to cwd

### LINKED ISSUE #61: Refactor cd to cwd?
We use cd for current directory, but in CLI it means change directory. Maybe we should switch to cwd. 
For convenience, both should be available for some versions using a deprication warning.

### LINKED ISSUE #121: Avoid retranslation upon setting the same model
DymolaAPI translates again if you set the same model_name again. This is not wanted.

### PR DESCRIPTION
**Pull request by FWuellhorst**
Closes #121 
Closes #61

### CODE REVIEW

**Comment by FWuellhorst** [comment]
Also, I added the option to only load certain variables from mats, which speeds thinks up by factors and reduces memory usage. Moreover, pathlib.Path is used for `working_directory` as a default, still allowing strings.

File: ebcpy/simulationapi/__init__.py [hunk 1]
```diff
@@ -478,20 +491,39 @@ def _update_model(self):
...
 
-    def set_cd(self, cd):
+    def set_working_directory(self, working_directory):
         """Base function for changing the current working directory."""
-        self.cd = cd
+        self.working_directory = working_directory
 
     @property
-    def cd(self) -> str:
+    def working_directory(self) -> str:
```
Comment (FelixStege) [inline ebcpy/simulationapi/__init__.py:499]: The return type will always be `Path`, right?

File: ebcpy/simulationapi/__init__.py [hunk 2]
```diff
@@ -478,20 +491,39 @@ def _update_model(self):
...
+        self.working_directory = cd
+
+    @property
+    def cd(self) -> str:
+        warnings.warn("cd was renamed to working_directory in all classes. "
+                      "Use working_directory instead instead.", category=DeprecationWarning)
+        return self.working_directory
 
     @cd.setter
     def cd(self, cd: str):
```
Comment (FelixStege) [inline ebcpy/simulationapi/__init__.py:523]: Does setting a string (and not a `Path`) here mess up anything down the line?

File: ebcpy/utils/__init__.py [hunk 3]
```diff
@@ -4,18 +4,20 @@
...
+                 working_directory: Union[Path, str] = None,
                  level=logging.DEBUG):
     """
     Setup an class or module specific logger instance
     to ensure readable output for users.
 
     :param str name:
         The name of the logger instance
-    :param str cd:
+    :param str working_directory:
```
Comment (FelixStege) [inline ebcpy/utils/__init__.py:20]: Add `Path` as additional type for `working_directory` to doc string

**Review by FelixStege — APPROVED** [review]

File: ebcpy/optimization.py [hunk 4]
```diff
@@ -40,14 +42,17 @@ class Optimizer:
     # Can be used, but will enlarge runtime
     _obj_his = []
 
-    def __init__(self, cd=None, **kwargs):
+    def __init__(self, working_directory: Union[Path, str] = None, **kwargs):
         """Instantiate class parameters"""
-        if cd is None:
-            self._cd = None
+        if working_directory is None and "cd" in kwargs:
+            warnings.warn("cd was renamed to working_directory in all classes. Use working_directory instead.", category=DeprecationWarning)
```
[the remark this answers is not in the document]
Comment (FWuellhorst) [reply ebcpy/optimization.py:48]: This is just backwards compatibility, cd is converted to workings_directory

**Comment by FWuellhorst** [comment]
@FelixStege thanks for the comments, I fixed them all!

**Pull request merged**